### ISSUE TITLE
Parsing an attribute without a value can cause code confusion

### ISSUE THREAD

**Issue by Bulandent** [xmldom/xmldom#486]
**Describe the bug**

Something went wrong when I parsed the following code。

```template
<template>
  <view>
    <image lazy />
    <image></image>
  </view>
</template>
```

The above code is parsed into something like this：

```template
<template>
  <view>
    <image lazy="lazy">
    <image/>
  
</image></view></template>
```

**To Reproduce**
https://stackblitz.com/edit/js-xmldom-template-cyojup?file=index.js

**Expected behavior**

```template
<template>
  <view>
    <image lazy="lazy"/>
    <image/>
  </view>
</template>
```

**Runtime & Version:**
xmldom version:  0.8.6
runtime version: 0.8.6
other related software and version:  

**Additional context**

I tried to solve the problem myself, until I changed the code in `@xmldom/xmldom/lib/sax.js` in such a way that it could parse properly。

```diff
case '/':
	switch(s){
	case S_TAG:
		el.setTagName(source.slice(start,p));
	case S_ATTR_END:
	case S_TAG_SPACE:
	case S_TAG_CLOSE:
		s =S_TAG_CLOSE;
		el.closed = true;
	case S_ATTR_NOQUOT_VALUE:
	case S_ATTR:
-	// case S_ATTR_SPACE:
		break;
+	case S_ATTR_SPACE:
+		el.closed = true;
+		break;
	//case S_EQ:
	default:
		throw new Error("attribute invalid close char('/')") // No known test case
	}
	break;
```
Finally, I put up the [pull request](https://github.com/xmldom/xmldom/pull/485/commits/83205ebed1d19ec886274269b90180519c45becd), I don't know whether my solution is right or not, so you can check it when you are free. 
Thanks.

**Comment by karfau** [comment]
Thank you for reporting, filing the bug and even the PR.
Let's see how the tests think about those changes.

**Comment by karfau** [comment]
Fixes are available in 0.7.10 and 0.8.7 releases

**Comment by karfau** [comment]
FYI: Just added the `xml:not well-formed` label since this is only valid HTML, not XML according to the specs and the `DOMParser` in browser will refuse to parse all of it:

![image](https://user-images.githubusercontent.com/135657/229084687-a1718fe1-34a1-4095-a9f0-ac71ca03b136.png)

Even when using `text/html` as a mimeType the behavior is different in the browser (where template tags end up in the `head` and it's content is inside a `DocumentFragment`):

![image](https://user-images.githubusercontent.com/135657/229084984-f33ab31b-72f4-4867-be88-401969cd6412.png)


compared to xmldom, which doesn't wrap html with a `html`, `body` and `head`and doesn't have any special handling of the `template` tag.
